Libplanet/ByteUtil.cs

```diff
@@ -90,39 +95,38 @@ public static ImmutableArray<byte> ParseHexToImmutable(string hex)
         /// Renders a hexadecimal string from a <see cref="byte"/> array.
         /// </summary>
         /// <param name="bytes">A <see cref="byte"/> array to renders
-        /// the corresponding hexadecimal string.  It must not be <see langword="null"/>.
+        /// the corresponding hexadecimal string.
         /// </param>
         /// <returns>A hexadecimal string which encodes the given
         /// <paramref name="bytes"/>.</returns>
-        /// <exception cref="ArgumentNullException">Thrown when the given
-        /// <paramref name="bytes"/> is <see langword="null"/>.</exception>
         /// <seealso cref="Hex(in ImmutableArray{byte})"/>
         /// <seealso cref="ParseHex(string)"/>
         [Pure]
         public static string Hex(byte[] bytes)
         {
-            if (bytes == null)
+            char[] chars = new char[bytes.Length * 2];
+            for (int i = 0; i < bytes.Length; i++)
             {
-                throw new ArgumentNullException(nameof(bytes));
+                chars[i * 2] = _hexCharLookup[bytes[i] >> 4];
+                chars[i * 2 + 1] = _hexCharLookup[bytes[i] & 0xf];
             }
 
-            string s = BitConverter.ToString(bytes);
-            return s.Replace("-", string.Empty).ToLower(CultureInfo.InvariantCulture);
+            return new string(chars);
         }
 
         /// <summary>
         /// Renders a hexadecimal string from a <see cref="byte"/> array.
         /// </summary>
         /// <param name="bytes">A <see cref="byte"/> array to renders
-        /// the corresponding hexadecimal string.  It must not be <see langword="null"/>.
+        /// the corresponding hexadecimal string.
         /// </param>
         /// <returns>A hexadecimal string which encodes the given
         /// <paramref name="bytes"/>.</returns>
         /// <seealso cref="Hex(byte[])"/>
         /// <seealso cref="ParseHexToImmutable(string)"/>
         [Pure]
         public static string Hex(in ImmutableArray<byte> bytes) =>
-            bytes.IsDefaultOrEmpty ? string.Empty : Hex(bytes.ToArray());
+            Hex(bytes.IsDefaultOrEmpty ? new byte[0] : bytes.ToArray());
```

Comment:
Probably slower, but thought it was just a tad bit more consistent. 🙃

Reply:
The end result is the same, but this indicates `default` or empty `ImmutableArray<byte>` will be both treated as an empty `byte` array, instead of the end result simply being `string.Empty`. Not a big deal, but still this is reliant upon how `Hex(byte[])` would behave.